backend/app/api/executions.py

```
@@ -0,0 +1,137 @@
+        if workflow_id:
+            query = query.filter(Execution.workflow_id == workflow_id)
+        
+        if status:
+            query = query.filter(Execution.status == status)
+        
+        executions = query.order_by(Execution.created_at.desc()).offset(skip).limit(limit).all()
+        return [ExecutionResponse.from_orm(execution) for execution in executions]
+        
+    except Exception as e:
+        logger.error(f"Error listing executions: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to list executions: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)